### ISSUE TITLE
MissingReferenceException: The object of type 'CastTargetController' has been destroyed but you are still trying to access it.

### ISSUE THREAD

**Issue by nigelmerle** [michaelday008/AnyRPGCore#74]
I reinstalled unity many times this week and even tested on different versions, i made sure that I did not install any package from the package manager and imported anyrpg with its default settings.

Lets start again:
-First you advised me to install unity free which i did  okay
-Second I copied anyrpg files to unity project folder , then I opened the project 
-I followed your entire tutorial, i even have the screenshots that you sent to me in my email
-First thing I did when the project opened I went to project settings to notice that unity have reset all the project settings back to default which in plain means that I had to recheck the .net framewok.
-then i imported the textmeshpro
-Then what i did i imported the uma from the unity store
-When all of it is installed in the project I restarted unity okay
-After I restarted unity I opened the build settings to make sure that all the scene files were in there
-Then what I did, I opened your new game wizard window and clicked on the feature demo game, so far so good everything works okay
-Then I wanted to test out your new controller, what I did I opened the defaultumaplayerunit.asset and then i checked the rotation checkbox 'only' , thats it
-Then when I compile the game, I press play, I get the cast controller target was destroyed error in unity
- In plain I understood that unity did not refresh the whole project in the scene view after i modified the defaultumaplaterunit.asset  in the inspector...i even tried to create a new empty scene in unity and open your create a game wizard and relaunch the anyrpg feature demo game again but the script were still not updated in the scene view
- I went to preference in unity to verify that autorefresh is checked on, it is checked on but unity somehow is not able to refresh the project in the scene view to reflect the changes 
- In the end I chose to close unity and reopen to see that the new changes in the defaultumaplayerunit.asset was applied successfully in the editor, everything works well so far okay
- what I am complaining about is the fact that I have to restart unity everytime after a slightest change in your scripts , its been two days of waiting for me : unity busy at, hold on..etc.  
-can you investigate with unity why this the defaultumaplayerunit.asset is not refreshing or maybe there is a fault in your coding practice because unity is not reflecting  the whole changes in the entire project..I dont know, im just saying, no offence...We are only trying to make it work. I thought i was going crazy with the unity problems but when I checked on internet many people are having these problems after they change scripts values in the inspector. Unity doesnt work as it is supposed to do.

For your personal information: I am on an acer predator with 16gb of memory (it is quite a powerful pc to handle such a small project) and my unity  of course is installed on a ssd. I bought this computer this year.. I usually overclock my pc when I open unity to see if there is any improvement in the loading times, but it does nothing to improve the reimport speed. Unity does its own thing, it will take all the time to think and ignore overclocking or memory increase. I was on 8gb ram and I increased to 16 gb, i dont see any change in unity opening speed

how you can reproduce that error ? i dont know , perhaps try to download anyrpg as a zip on github, thats what i did and install on a new machine and try to follow my steps...i downloaded it as a zip on github,...we dont have the same hardware so we might not get the same results

**Comment by nigelmerle** [comment]
I am still testing...After reimport there is no more errors...however if I hit the play button a second time the the error will show up again. So I can compile the project only once, maybe there is a bug in the script itself

![Screenshot (40)](https://user-images.githubusercontent.com/17375811/147242709-c8ee64a5-fa24-4993-ab9d-69430e83ea79.png)
r

**Comment by michaelday008** [comment]
Please re-edit the initial description to remove anything that is not an exact reproduction step.  That means remove everything related to youtube, etc.

**Comment by nigelmerle** [comment]
im sorry about it, i was in a hurry, i just wrote what was on my mind but i see you did it yourself thanks

**Comment by nigelmerle** [comment]
its been a week of struggling for me, i do not sleep everyday because of these issues

**Comment by nigelmerle** [comment]
did you mean about the title of the issue that i wrote?

**Comment by nigelmerle** [comment]
or are you asking me to remove my comment from youtube ?

**Comment by nigelmerle** [comment]
ok i thought i find what you mean, it was on the first paragraph of the decriptions

**Comment by nigelmerle** [comment]
Right now I am trying to reinstall visual studio, i read on the internet that they have an option in their settings for refreshing with unity projects..I will let you know in a moment if it worked

**Comment by nigelmerle** [comment]
i also tried to remove all the read only permission from my unity project folders

**Comment by michaelday008** [comment]
A Few things:

1.
I'm not sure what you mean by recheck the .net framework.  Can you provide a screenshot?

2.
>>Then I wanted to test out your new controller, what I did I opened the defaultumaplayerunit.asset and then i checked the rotation checkbox 'only' , thats it".

If you are using a gamepad, this is not necessary.  When you are playing, you can open the settings menu and go to controls and turn on gamepad mode.

3.
>>In plain I understood that unity did not refresh the whole project in the scene view after i modified the defaultumaplaterunit.asset in the inspector...i even tried to create a new empty scene in unity and open your create a game wizard and relaunch the anyrpg feature demo game again but the script were still not updated in the scene view.

I'm not sure what you mean by this.  The DefaultUMAPlayerUnit does not exist in the scene.  It's a scriptable object.  There is no reason for the scene to refresh anything after you make a change to it.

4.
In the screenshot you provided, the game has already exited.  Can you ensure that "Error Pause" is selected in the console view and send another screenshot with play mode still active?  I would need to see what the heirarchy looks like at the moment the error happens.  I specifically need to see the following things (circled in my screen shot)
A. The game manager is properly set under "DontDestroyOnLoad"
B. The CastTarget actually exists under the game manager
C. If you select the game manager, and scroll down past the settings until you find the CastTargettingManager, the link to the CastTarget is not null

This is what it should look like:

![image](https://user-images.githubusercontent.com/49324924/147296449-15307bcb-cc33-41c8-924c-61d01c79393e.png)

**Comment by michaelday008** [comment]
Regarding the issue with Unity being slow, have you added windows defender exclusions to your Unity directories?  You should have an exclusion for the folder where Unity is installed, as well as the folder that will contain all your Unity projects.  Here's an example:
![image](https://user-images.githubusercontent.com/49324924/147297860-cff07727-a1bd-4b83-987b-c19f05bd5ccb.png)

**Comment by michaelday008** [comment]
It looks like you didn't uninstall your cracked version of Unity Pro properly.

Take a look at the menu bar on my Unity.  Notice it says "Personal" at the end of the title.  In the screenshot you sent, it does not say Personal.

I think you may have skipped the step where you are supposed to delete the license file.

Look through the answers on this page about other people having problems with cracked versions, and ensure you delete the license file like they did.

https://gamedev.stackexchange.com/questions/164670/unity-editor-wont-open

![image](https://user-images.githubusercontent.com/49324924/147305297-47ed2da1-7b59-49e0-a32d-5bed06f820c6.png)

**Comment by SickaGamer** [comment]
If you are using a torrent version of Unity, you might as well give up.  No
one wants to help out pirates.

**Comment by nigelmerle** [comment]
I am currently reinstalling my machine give me a minute, you wrote to me at 3 am.
I dont know what you mean by no one wants to help pirates, im not a pirate, i dont even know how to be one, this is a personal accusation but you can keep that for yourself, this is what you believe, personally i dont believe that im a bad person. The bad person are the ones who are doing human slavery on earth. Well We are talking about issues with your program,what you you said is totally out of scope, can we focus on what really matters. You must be glad that i am showing some interests in your project and that i am pointing out those errors for free to you, im not being paid for that. when people will come across the errors  most of the time they will not tell you anything, they will just drop it. You cam call me a sick gamer, We are both sick gamers by the way because else we wouldn't be here trying and trying to make games.
Id be happy if we can stop picking on each other and be more professional. Im writing you in a second.



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by nigelmerle** [comment]
Robinhood was a good pirate though because he helped the people around him

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by nigelmerle** [comment]
Internet and the darkweb as the Robinhood helped people to survive in many ways..there are things far worse than this happening around us in that fake democracy and the government have legitimately been given the freedom to opress humanity with laws and rules that never existed before.They say that they work for us but infact they dont.



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by nigelmerle** [comment]
Hello this is to let you know that I think I finally fixed it. 

Apparently I had a corrupted visual studio 2019 on my pc that I downloaded from unity website. What I did, I installed visual studio 2022 community instead from microsoft.
In visual studio interface I check both these packages :  desktop development with c++ and  games development with unity.
  In the unity project settings, i went to preference and add the visual studio 2022 to it  and hit regenerate project files...here are my settings in unity. Im not quite sure why suddenly it decides to work, i suppose something wasnt installed properly. Thats strange because I reinstalled unity several times. 

![Screenshot (186)](https://user-images.githubusercontent.com/17375811/147314361-8d0062f3-5139-4e8d-b76f-7a2205a92812.png)
